docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,382 @@
+
+    - a device was provisioned (added to core metadata)
+    - a service was stopped
+    - service configuration has changed
+
+    Note CPE should not be confused with core data Events.  Core data Events represent a collection (one or more) of sensor/device readings.  Core data Events represent sensing of some measured state of the physical world (temperature, vibration, etc.).  CPE represents the detection of some happening inside of the EdgeX software.
+
+!!! Info
+    Information in *italics* indicates topics for discussion without suggested proposal at this time.
+
+## Context
+
+[System Management services](../../../microservices/system-management/Ch_SystemManagement.md) (SMA and executors) currently provide a limited set of “metrics” to requesting clients (3rd party applications and systems external to EdgeX).  Namely, it provides requesting clients with service CPU and memory usage; both metrics about the resource utilization of the service itself versus metrics that are about what is happening inside of the service.  Arguably, the current system management metrics can be provided by the container engine and orchestration tools (example: by Docker engine) or by the underlying OS tooling.
+
+!!! Info
+    The SMA has been deprecated and will be removed in a future, yet named, release.
+
+Going forward, users of EdgeX will want to have more insights – that is more metrics telemetry – on what is happening directly in the services and the tasks that they are preforming.  In other words, users of EdgeX will want more telemetry on service activities to include:
+- sensor data collection (how much, how fast, etc.)
+- command requests handled (how many, to which devices, etc.)
+- sensor data transformation as it is done in application services (how fast, what is filtered, etc)
+- sensor data export (how much is sent, how many exports have failed, etc. )
+- API requests (how often, how quickly, how many success versus failed attempts, etc.)
+- bootstrapping time (time to come up and be available to other services)
+- activity processing time (amount of time it takes to perform a particular service function - such as respond to a command request)
+
+### Definitions
+
+**Metric (or telemetry) data** is defined as the count or rate of some action, resource, or circumstance in the EdgeX instance or specific service.  Examples of metrics include:
+
+- the number of EdgeX Events sent from core data to an application service via message bus (or via device service to application service in Ireland and beyond)
+- the number of requests on a service API
+- the average time it takes to process a message through an application service
+- The number of errors logged by a service
+
+The collection and dissemination of metric data will require internal service level instrumentation (relevant to that service) to capture and send data about relevant EdgeX operations.  EdgeX does not currently offer any service instrumentation.

```

Comment:
Isn't there some measure of instrumentation today in order to support the albeit limited metrics available via SMA?